src/record_merge.jl

```diff
@@ -0,0 +1,12 @@
+"""
+    record_merge(record; fields_to_merge...)
+
+Return a new `AbstractRecord` with the same schema as `record` by merging the fields of
+`record` and `fields_to_merge` together using `Tables.rowmerge`.
+"""
+function record_merge(record::AbstractRecord; fields_to_merge...)
```

Comment:
Doesn't have to be part of this PR but it could be nice to have an additional method that works analogously to `rowmerge`, then this method could simply call that one:
```julia
function record_merge(record::AbstractRecord, other, others...)
    R = record_type(schema_version_from_record(record))
    return R(Tables.rowmerge(record, other, others...))
end

record_merge(record::AbstractRecord; fields...) = record_merge(record, (; fields...))
```

Reply:
I did notice that other method but decided to not introduce it as I personally didn't have a need for it. Also, we may want to restrict other rows to the same `AbstractRecord` subtype and avoid updating non-missing values with `missing`.

Reply:
The `missing` thing seems especially problematic as the keyword method allows you to set a field to `missing` but if we did the same with `record_merge(record::T, other::T)` you'll end up just with `other`.